scrapy/spidermiddlewares/robotstxt.py

```diff
@@ -0,0 +1,20 @@
+_start_requests_processed = object()
+
+
+class RobotsTxtSpiderMiddleware:
+    @classmethod
+    def from_crawler(cls, crawler):
+        return cls(crawler)
+
+    def __init__(self, crawler):
+        self._send_signal = crawler.signals.send_catch_log
+
+    def process_start_requests(self, start_requests, spider):
+        # Mark start requests and reports to the downloader middleware the
+        # number of them once all have been processed.
+        count = 0
+        for request in start_requests:
+            request.meta["is_start_request"] = True
```

Comment:
should we make it a general Scrapy feature somehow, including the "start_requests_processed" signal? Or maybe even "on_start_request" signal in addition to that, so that we don't need a meta flag.